indico/modules/events/contributions/util.py

```diff
@@ -244,6 +244,21 @@ def get_contributions_for_person(event, person, only_speakers=False):
             .all())
 
 
+def get_contributions_for_user(event, user):
+    """Get all contributions for a user in the given event.
+
+    Gets all contributions where the user is a primary/secondary author, speaker or a submitter.
+    """
+    condition = db.or_(EventPerson.user == user,
+                       Contribution.acl_entries.any(db.and_(ContributionPrincipal.has_management_permission('submit'),
+                                                            ContributionPrincipal.user == user)))
+    return (Contribution.query.with_parent(event)
+            .join(ContributionPersonLink)
+            .join(EventPerson)
```

Comment:
```suggestion
            .join(EventPerson, outer=True)
```

otherwise the whole query won't return a user with no matching EventPerson (but please test that case just in case)